kraken-std/src/kraken/std/python/buildsystem/maturin.py

```diff
@@ -78,86 +52,51 @@ class MaturinZigTarget:
     zig_features: Collection[str] = ()
 
 
-class MaturinPythonBuildSystem(PoetryPythonBuildSystem):
-    """A maturin-backed version of the Poetry build system, that invokes the maturin build-backend.
-    Can be enabled by adding the following to the local pyproject.yaml:
-    ```toml
-    [tool.poetry.dev-dependencies]
-    maturin = "0.13.7"
-
-    [build-system]
-    requires = ["maturin>=0.13,<0.14"]
-    build-backend = "maturin"
-    ```
-    """
-
-    name = "Maturin"
-
-    def __init__(self, project_directory: Path) -> None:
-        super().__init__(project_directory)
+class _MaturinBuilder:
+    def __init__(self, entry_point: str) -> None:
+        self._entry_point = entry_point
         self._default_build = True
-        self._zig_targets: list[MaturinZigTarget] = []
+        self._zig_targets: Collection[MaturinZigTarget] = []
 
     def disable_default_build(self) -> None:
         self._default_build = False
 
-    def enable_zig_build(
-        self,
-        targets: Collection[str | MaturinZigTarget] = (),
-        features: Collection[str] = (),
-        macos_sdk_root: Path | None = None,
-    ) -> None:
+    def enable_zig_build(self, targets: Collection[MaturinZigTarget]) -> None:
         """
         :param targets: Collection of MaturinTargets to cross-compile to using zig.
-        :param features: deprecated, do not use.
-        :param macos_sdk_root: deprecated, do not use.
         """
-        self._zig_targets.clear()
-        for target in targets:
-            if isinstance(target, str):
-                self._zig_targets.append(
-                    MaturinZigTarget(target=target, macos_sdk_root=macos_sdk_root, zig_features=features)
-                )
-            else:
-                self._zig_targets.append(target)
-
-    # PythonBuildSystem
-
-    def get_pyproject_reader(self, pyproject: Pyproject) -> MaturinPyprojectHandler:
-        return MaturinPyprojectHandler(pyproject)
-
-    def get_managed_environment(self) -> ManagedEnvironment:
-        return MaturinManagedEnvironment(self.project_directory)
+        self._zig_targets = targets
 
-    def update_pyproject(self, settings: PythonSettings, pyproject: Pyproject) -> None:
-        super().update_pyproject(settings, pyproject)
-        handler = self.get_pyproject_reader(pyproject)
-        handler.synchronize_project_section_to_poetry_state()
-
-    def build(self, output_directory: Path, as_version: str | None = None) -> list[Path]:
+    def build(
+        self,
+        output_directory: Path,
+        as_version: str | None,
+        get_pyproject_reader: Callable[[Pyproject], PyprojectHandler],
+        project_directory: Path,
```

Comment:
I know they are only used in `build`, but would it make sense to store `get_pyproject_reader` and `project_directory` (potentially `output_directory` too) in the constructor to simplify `build`'s signature?